src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -1486,7 +1492,10 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 		// reset cwd if it has changed, so file based url paths can be resolved
 		const cwd = await this.getCwd();
 		if (cwd && this._linkManager) {
-			this._linkManager.processCwd = cwd;
+			if (this._linkManager.processCwd !== cwd) {
```

Comment:
@Tryiar suggeseted link manager could listen to the change property event event instead of on _updateProcessCwd (which would actually be better called _refresh)